auxx/src/Command/DumpBlockchain.hs

```diff
@@ -39,84 +43,171 @@ import           Mode (MonadAuxxMode)
+
+    fastForwardEpoch
+        :: (MonadAuxxMode m, MonadIO m)
+        => BlockHeader -> m (Maybe BlockHeader)
+    fastForwardEpoch start = do
+        let epochIndex = start ^. epochIndexL
+        logInfo ("Skipping epoch "+|epochIndex|+"")
+        maybePrev <- skipEpoch $ headerHash start
+        case maybePrev of
+            Nothing -> pure Nothing
+            Just prev -> DB.getHeader prev >>= \case
+                Just header -> pure (Just header)
+                Nothing     -> do
+                    logWarning ("DB doesn't contain block "+|prev|+"")
+                    pure Nothing
+
+-- | Run an endless loop that will be dumping epochs as soon as they become
+-- stable.
+dumpBlockchainLoop
+    :: (MonadAuxxMode m, MonadIO m)
+    => FilePath
+    -> m ()
+dumpBlockchainLoop outFolder = forever $ do
+    -- Get the last stable epoch – i.e. one that won't ever change or be
+    -- rolled back. To do that, we subtract 2K+2 slots from the last block's
+    -- slot and take the *previous* epoch (2K+2 instead of 2K+1 because the
+    -- last block might be a genesis block and then it would correspond to
+    -- slot N:0 even though the block N:0 wasn't issued yet).
+    lastSlot <- flattenEpochOrSlot . getEpochOrSlot <$> DB.getTipHeader
+    let mbStableEpoch = do
+            stableSlot <- subtractMay (getSlotCount (slotSecurityParam + 2))
+                                      lastSlot
+            subtractMay 1 (unflattenSlotId stableSlot ^. epochIndexL)
+    -- Look at the filenames of existing files to determine the last epoch
+    -- which we have already dumped.
+    mbLastDumpedEpoch <-
+        maximumMay . mapMaybe parseEpochFileName <$>
+        liftIO (listDirectory outFolder)
+    -- Dump all epochs in the interval [last dumped + 1; stable], or all
+    -- epochs until 'stable' if there are no dumped epochs.
+    case (mbStableEpoch, mbLastDumpedEpoch) of
+        (Nothing, _) ->
+            logInfo "No epoch is stable yet; not going to dump anything"
+        (Just stableEpoch, Nothing) -> do
+            logInfo ("The last stable epoch is "+|stableEpoch|+"")
+            dumpBlockchain Nothing (Just stableEpoch) outFolder
+        (Just stableEpoch, Just lastDumpedEpoch)
+            | stableEpoch <= lastDumpedEpoch ->
+                  logInfo ("Not going to dump anything; the last " <>
+                           "stable epoch ("+|stableEpoch|+") is " <>
+                           "already dumped")
+            | otherwise -> do
+                  logInfo ("The last stable epoch is "+|stableEpoch|+"")
+                  dumpBlockchain (Just (lastDumpedEpoch + 1))
+                                 (Just stableEpoch)
+                                 outFolder
+    -- Wait before repeating. Without the delay, we'll spend all the time
+    -- doing IO. (A one-hour delay before a fresh epoch becomes available
+    -- seems fine, given that epochs last several days.)
+    delay (hour 1)
```

Comment:
Instead of polling, this could probably be hooked into `BListener`. However, I don't think the benefit of not having to wait for half an hour (on average) for the new epoch to be served is worth rewriting all this code.